### PR TITLE
[No merge] TF integration testing

### PR DESCRIPTION
**Pull request by LysandreJik**
Adds integrationt tests for BERT, ELECTRA and Longformer to ensure that PR such as https://github.com/huggingface/transformers/pull/7605 do not impact the current state of models. RoBERTa not done because it's already done.

Patches a bug with the `ElectraForPreTraining` when batch size = 1

### CODE REVIEW

File: tests/test_modeling_tf_bert.py [hunk 1]
```diff
@@ -328,3 +328,27 @@ def test_custom_load_tf_weights(self):
         self.assertEqual(sorted(output_loading_info["unexpected_keys"]), ["mlm___cls", "nsp___cls"])
         for layer in output_loading_info["missing_keys"]:
             self.assertTrue(layer.split("_")[0] in ["dropout", "classifier"])
+
+
+class TFBertModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_masked_lm(self):
+        model = TFBertForPreTraining.from_pretrained("lysandre/tiny-bert-random", use_cdn=False)
```
Comment (LysandreJik) [inline tests/test_modeling_tf_bert.py:336]: Will remove the `use_cdn=False` tomorrow once the CDN caching has caught up.

```suggestion
        model = TFBertForPreTraining.from_pretrained("lysandre/tiny-bert-random")
```

File: tests/test_modeling_tf_electra.py [hunk 2]
```diff
@@ -248,3 +248,19 @@ def test_model_from_pretrained(self):
         for model_name in ["google/electra-small-discriminator"]:
             model = TFElectraModel.from_pretrained(model_name)
             self.assertIsNotNone(model)
+
+
+class TFElectraModelIntegrationTest(unittest.TestCase):
+    @slow
+    def test_inference_masked_lm(self):
+        model = TFElectraForPreTraining.from_pretrained("lysandre/tiny-electra-random", use_cdn=False)
```
Comment (LysandreJik) [inline tests/test_modeling_tf_electra.py:256]: Will remove the `use_cdn=False` tomorrow once the CDN caching has caught up.

```suggestion
        model = TFElectraForPreTraining.from_pretrained("lysandre/tiny-electra-random")
```

File: tests/test_modeling_tf_longformer.py [hunk 3]
```diff
@@ -548,3 +548,25 @@ def test_inference_masked_lm_long(self):
         tf.debugging.assert_near(tf.reduce_mean(loss), expected_loss, rtol=1e-4)
         tf.debugging.assert_near(tf.reduce_sum(prediction_scores), expected_prediction_scores_sum, rtol=1e-4)
         tf.debugging.assert_near(tf.reduce_mean(prediction_scores), expected_prediction_scores_mean, rtol=1e-4)
+
+    @slow
+    def test_inference_masked_lm(self):
+        model = TFLongformerForMaskedLM.from_pretrained("lysandre/tiny-longformer-random", use_cdn=False)
```
Comment (LysandreJik) [inline tests/test_modeling_tf_longformer.py:554]: Will remove the `use_cdn=False` tomorrow once the CDN caching has caught up.

```suggestion
        model = TFLongformerForMaskedLM.from_pretrained("lysandre/tiny-longformer-random")
```

**Comment by jplu** [comment]
@LysandreJik Is this PR done with your last changes in the tests?

**Comment by LysandreJik** [comment]
Ah, I had forgotten about this. I'll rebase and ping you for review

**Comment by LysandreJik** [comment]
should be good for review @jplu

**Review by jplu — APPROVED** [review]
Just a tiny question, are these ITs the same than the PyTorch ones?

Looks perfect to me! Thank you very much for taking care of this!!

**Comment by LysandreJik** [comment]
They're not the same as they don't rely on the full checkpoints but on some random tiny ones, to make the CI faster.
It does test every same aspect, however: the weights loading, the full inference, the expected results.

**Pull request merged**